### DISCUSSION TITLE
Dataset size and learning rate

### DISCUSSION THREAD

**Discussion by denisvorotyntsevbidease** [catboost/catboost#2573] [General]
Problem:

I train a CatBoost model using the training part of the dataset to estimate the validation error. Then, I retrain the model using the full data (training + validation). Do I need to change the hyperparameters of CatBoost to improve performance (number of trees, learning rate, etc.)? 

I heard a recommendation to increase the learning rate by the square root of the increased sample size (e.g., 30% more data -> increase the learning rate by 1.3^0.5), but I haven't found any theoretical justifications yet. Any recommendations? Is it a heuristic? 

catboost version:
Operating System:
CPU:
GPU:

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
it depends but you may have a look at how catboost chooses learning rate
this code is based on experiments that we ran on datasets from a number of open competitions
https://github.com/catboost/catboost/blob/2c0420a55ffece7c532a928b427f54acf70dae25/catboost/libs/train_lib/options_helper.cpp#L117
